marconi-chain-index/src/Marconi/ChainIndex/Types.hs

```diff
@@ -142,13 +161,15 @@ newtype TxIndexInBlock = TxIndexInBlock Word64
     )
 
 -- | Common configuration required to run indexers
-data RunIndexerConfig = RunIndexerConfig
-  { _runIndexerConfigTrace :: MarconiTrace IO
+data RunIndexerConfig m = RunIndexerConfig
+  { _runIndexerConfigTrace :: MarconiTrace m
```

Comment:
Maybe you could leave this with `IO` and use `liftIO` to run it? It seems a little odd to have the config parameterized. I feel like it should be possible to use the same config with any `MonadIO`. The version of the logging functions that operate in the `ReaderT` could do the `liftIO` so that the caller doesn't have to.

Reply:
I'll try that, thanks for pointing it out!